ccl/changefeedccl: TestChangefeedNemeses failed

ccl/changefeedccl.TestChangefeedNemeses [failed](https://tanzanite.cluster.engflow.com/invocation/485553d2-82cc-4eb4-865f-1928c5443d65) on master @ [e174990dd5015a38b1e9bc6723f270dbe647c3a3](https://github.com/cockroachdb/cockroach/commits/e174990dd5015a38b1e9bc6723f270dbe647c3a3):


```
=== RUN   TestChangefeedNemeses
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedNemeses704209802
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestChangefeedNemeses/webhook
    helpers_test.go:853: making server as system tenant
    helpers_test.go:928: making webhook feed factory
    nemeses_test.go:32: random seed: -8324793835391516166
panic: use of Span after Finish. Span: changefntr. Finish previously called at: <stack not captured. Set debugUseAfterFinish>

goroutine 96003 [running]:
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish(0xbd8d640?)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:182 +0x114
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).RecordingType(0xc00ba18000)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:442 +0x18
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).IsVerbose(...)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:450
github.com/cockroachdb/cockroach/pkg/util/log.getSpanOrEventLog({0x7bbb530, 0xc008d45540})
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:92 +0x34
github.com/cockroachdb/cockroach/pkg/util/log.logfDepthInternal({0x7bbb530, 0xc008d45540}, 0x2, 0x3, 0x0, 0x3?, {0x64f156c, 0x21}, {0xc005927c70, 0x1, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:104 +0x4b4
github.com/cockroachdb/cockroach/pkg/util/log.logfDepth(...)
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:39
github.com/cockroachdb/cockroach/pkg/util/log.Errorf(...)
	github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/util/log/log_channels_generated.go:654
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*batchingSink).runBatchingWorker(0xc0117c8e00, {0x7bbb530, 0xc008d45540})
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/batching_sink.go:404 +0x58f
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.makeBatchingSink.func1({0x7bbb530?, 0xc008d45540?})
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/batching_sink.go:514 +0x2a
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.makeBatchingSink.Group.GoCtx.func2()
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x22
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75 +0x56
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 95979
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:72 +0x96
```
<p>Parameters: <code>attempt=1</code>
, <code>run=17</code>
, <code>shard=13</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedNemeses.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-33359

--- a/pkg/ccl/changefeedccl/batching_sink.go
+++ b/pkg/ccl/changefeedccl/batching_sink.go
@@ -20,7 +20,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/intsets"
-	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/retry"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 )
@@ -399,11 +398,12 @@ func (s *batchingSink) runBatchingWorker(ctx context.Context) {
 	for {
 		select {
 		case req := <-s.eventCh:
-			if err := s.pacer.Pace(ctx); err != nil {
-				if pacerLogEvery.ShouldLog() {
-					log.Errorf(ctx, "automatic sink batcher pacing: %v", err)
-				}
-			}
+			// Swallow pacer error -- it happens only if context is canceled,
+			// and that's handled below.
+			// TODO(yevgeniy): rework this function: this function should simply
+			// return an error, and not rely on "handleError".
+			// It's hard to reason about this functions correctness otherwise.
+			_ = s.pacer.Pace(ctx)
 
 			switch r := req.(type) {
 			case *rowEvent:

--- a/pkg/ccl/changefeedccl/event_processing.go
+++ b/pkg/ccl/changefeedccl/event_processing.go
@@ -13,7 +13,6 @@ import (
 	"hash"
 	"hash/crc32"
 	"runtime"
-	"time"
 
 	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval"
 	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdcevent"
@@ -36,10 +35,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/cockroachdb/errors"
 )
-
-// pacerLogEvery is used for logging errors instead of returning terminal
-// errors when pacer.Pace returns an error.
-var pacerLogEvery log.EveryN = log.Every(100 * time.Millisecond)
 
 // eventContext holds metadata pertaining to event.
 type eventContext struct {
@@ -328,9 +323,7 @@ func (c *kvEventToRowConsumer) ConsumeEvent(ctx context.Context, ev kvevent.Even
 	// unavailable. If there is unused CPU time left from the last call to
 	// Pace, then use that time instead of blocking.
 	if err := c.pacer.Pace(ctx); err != nil {
-		if pacerLogEvery.ShouldLog() {
-			log.Errorf(ctx, "automatic pacing: %v", err)
-		}
+		return err
 	}
 
 	schemaTimestamp := ev.KV().Value.Timestamp

